semgrep/tests/e2e/test_check.py

```diff
@@ -209,3 +209,29 @@ def test_max_memory(run_semgrep_in_tmp, snapshot):
         ),
         "error.txt",
     )
+
+
+def test_max_timeouts(run_semgrep_in_tmp, snapshot):
+    # Check that semgrep-core timeouts are properly handled
+
+    snapshot.assert_match(
+        run_semgrep_in_tmp(
+            "rules/long.yaml",
+            options=["--timeout", "1", "--max-timeouts", "1"],
```

Comment:
Probably worth adding a test that has max-timeouts > 1 and checking that it doesn't stop running rules until it hits max-timeouts. Can probably just create another rule file that duplicates the rule in long.yaml a bunch of times.

Reply:
Will do!